Paf 49 test cert 2

JHoldergov opened the pull request:
## What?
To access PRP1 we have to access the link via a secure way. So we have a self-signing certificate provided. 
## Why?
This is the most secure way for the services to talk to each other, we have set the NODE_AUTH=0 because as a self signing cert it does not have a recognised authority hence why we keep the following in the ims reolver deployment
```
        - name: NODE_OPTIONS
              value: "--use-openssl-ca"

            {{ if or (eq .KUBE_NAMESPACE .BRANCH_ENV) (eq .KUBE_NAMESPACE .UAT_ENV) }}
            - name: NODE_TLS_REJECT_UNAUTHORIZED
              value: "0"
```
We will need to add prod option in there.

## How?
So the best way for Node is to mount a config map as a persistent storage to store the cert then mount the config map as a volume in/etc/ss/certs and call it in the PAF app.

## Testing?
@shamiluwais had a look and will do some further verification that everything is as it is as we are currently facing a 504 and the logs are attached below

## Screenshots (optional)
![image](https://github.com/UKHomeOffice/paf/assets/125476151/281e1ac6-cf60-4a71-b25a-c884ef999174)
![image](https://github.com/UKHomeOffice/paf/assets/125476151/87e33d3a-8d65-4298-ae7f-4b40c890177a)


## Anything Else? (optional)
We are waiting for the error 504 to be resolved and then we can test but prior to this we did have a case id 

## Screenshots (optional)
## Anything Else? (optional)
let me know if you need anything - so as for the cert we have it stored securely in hof-services config as a base64 encoded cert and called it via the deploy.sh script

sulthan-ahmed committed:
PAF-50 Deploy ims-resolver

sulthan-ahmed committed:
PAF-50 add wsdl and endpoint as secrets

sulthan-ahmed committed:
PAF-50 deploy ims-resolver

sulthan-ahmed committed:
PAF-50 deploy ims-resolver

sulthan-ahmed committed:
PAF-50 deploy ims-resolver

sulthan-ahmed committed:
PAF-50 deploy ims-resolver

sulthan-ahmed committed:
PAF-50 deploy ims-resolver

sulthan-ahmed committed:
PAF-50 deploy ims resolver

sulthan-ahmed committed:
PAF-50 update ims resolver key names as part of k8

sulthan-ahmed committed:
Update README about IMS API docs

there is a new repo about the API documentation

sulthan-ahmed committed:
PAF-49 Test the private link

sulthan-ahmed committed:
WIP

sulthan-ahmed committed:
Deploy resolver that doesn't run the commands

* We will kube exec in and try to run the commands
via Node REPL

sulthan-ahmed committed:
PAF-49 change the ims-resolver to use curl

sulthan-ahmed committed:
WIP

sulthan-ahmed committed:
WIP

sulthan-ahmed committed:
PAF-49 deploy resolver to test private link

sulthan-ahmed committed:
WIP

sulthan-ahmed committed:
Set NODE_TLS_REJECT_UNAUTHORIZED 0 in kube file

sulthan-ahmed committed:
WIP check we can set env variable for TLS being 0

update image

sulthan-ahmed committed:
PAF-135 use ims-resolver image for private link

JHoldergov committed:
deploy tls cert - PAF-153

JHoldergov committed:
Pvt link add deploy branch

JHoldergov committed:
error message fix

JHoldergov committed:
self signed cert

JHoldergov committed:
redeploy

JHoldergov committed:
redeploy

JHoldergov committed:
mounting cert volume

JHoldergov committed:
deploy error

JHoldergov committed:
disable cert bypass

JHoldergov committed:
configmap trial

JHoldergov committed:
syntax fix

JHoldergov committed:
corrrection

JHoldergov committed:
error solve attempt

JHoldergov committed:
re-deployment

JHoldergov committed:
adjustments

JHoldergov committed:
timeout changes

JHoldergov committed:
adding config map

JHoldergov committed:
memory increase

JHoldergov committed:
syntax

JHoldergov committed:
config on node

TemitopeAyokuHO committed:
Added certificate

JHoldergov committed:
code changes for cert (trial)

JHoldergov committed:
syntax

JHoldergov committed:
untrusted source cert

JHoldergov committed:
change directory

JHoldergov committed:
dir correction

JHoldergov committed:
try two vol mounts

JHoldergov committed:
volume mounts testing

JHoldergov committed:
typo

JHoldergov committed:
testing branch 2

JHoldergov committed:
fast build CI

JHoldergov committed:
rebuild with yarn - redeploy

JHoldergov committed:
commenting out certs

JHoldergov committed:
vol mount deployment

JHoldergov committed:
alignment

JHoldergov committed:
alignment2

JHoldergov committed:
imsrepo-change

JHoldergov committed:
new ims image

JHoldergov committed:
new ims image

JHoldergov committed:
new image - 1520

JHoldergov committed:
new ims iiage 1533

JHoldergov committed:
new ims image 1559

jamiecarterHO committed:
Update to latest resolver image

jamiecarterHO committed:
Fix merge conflicts part two

jamiecarterHO committed:
Remove changes from server.js

jamiecarterHO committed:
Fix discrepancies between fv URLs across deployments and ingress (#122)
rebasing the branch

JHoldergov committed:
removed clear text cert

JHoldergov committed:
setting the cert in the hof config services

JHoldergov committed:
revert the test changes

JHoldergov committed:
add cert from hof cofig

JHoldergov committed:
cleaning up

JHoldergov committed:
memory re-adjust

JHoldergov committed:
merge conf

JHoldergov committed:
merge conf

JHoldergov committed:
indent error

JHoldergov committed:
Configmap1

JHoldergov committed:
cert mount

JHoldergov committed:
adding a cert mount without indent

sulthan-ahmed commented:
can’t approve til it’s tested

sulthan-ahmed commented:
Just fixed your PR, the headings weren't showing.  You had your backtiks in the wrong place

sulthan-ahmed commented on apps/paf/models/file-upload.js:
```diff
@@ -1,4 +1,3 @@
-/* eslint-disable node/no-deprecated-api */
```
is there a reason why `node/no-deprecated-api */` this was removed?

Rhodine-orleans-lindsay committed:
PAF 166:Highlight sections on menu navigation (#95)

- Add element ids to side-nav html
- Set current url and class list for navigation sections in assets/js/index.js
- Add css styling for highlighting navigation sections in assets/css/app.scss

Co-authored-by: muthrajan <144831064+muthrajan@users.noreply.github.com>

TemitopeAyokuHO committed:
PAF-129/ PAF-128 - Fix bug to add country from list and validate country textbox. (#96)

* PAF-128 & PAF-129 Added validation for phone and country textbox

[PAF- 129](https://collaboration.homeoffice.gov.uk/jira/browse/PAF-129) All 'Country' fields allow any text without the use of drop down list

[PAF-128](https://collaboration.homeoffice.gov.uk/jira/browse/PAF-128 URL's are being accepted in all free text fields

- added validation rules to all countries fields in field/index.js

- added dependent property to avoid error message when yes  is selected on another location

Tested manually

* add behaviour to unset values

jamiecarterHO committed:
rebase branch

* PAF-61 Update ims-resolver deployment to use branch image

* Deploy with image containing PAF-61 changes
* Update resolver configmap to use correct value
* Add Keycloak variables to ims resolver deployment env from secret

* PAF-61 Update resolver image to latest for ticket.

jamiecarterHO committed:
Update drone.yml and deploy.sh (#129)

* Add missing deploys to delete
* Alter get pr branch command to be more like other new repos

JHoldergov committed:
trigger build

JHoldergov committed:
rebased

JHoldergov committed:
rebased

JHoldergov committed:
rebasing

JHoldergov committed:
rebase

JHoldergov committed:
rebase

JHoldergov committed:
rebase

JHoldergov committed:
rebase

JHoldergov committed:
rebase

JHoldergov committed:
rebase

JHoldergov committed:
rebase

JHoldergov committed:
rebase

JHoldergov committed:
rebase

JHoldergov committed:
ims cert image

JHoldergov committed:
ims redeploy rebase

JHoldergov committed:
cert code rebase

JHoldergov committed:
file upload rebase
